docs/user/device-api-statuses.md

```diff
@@ -0,0 +1,126 @@
+| `Updating` | The device is in the process of updating to its device spec. | `deviceIsUpdating` |
+| `OutOfDate` | The device is not `UpToDate` and not `Updating`. | |
+| `Unkown` | The device's agent either never reported status or its last reported status was `Updating` and the device has been disconnected since. | |
+
+whereby the formal definition uses the following helper definitions:
+
+```golang
+// a device is assumed disconnected if its agent hasn't sent an update for the duration of a disconnectionTimeout
+deviceIsDisconnected := device.status.lastSeen + disconnectionTimeout < time.Now()
+
+// a device is not managed by a fleet if its owner field is unset
+deviceIsNotFleetManaged := len(device.metadata.owner) == 0
+
+// a device is updating when the agent sets the "Updating" condition to true
+deviceIsUpdating := device.status.conditions.updating == true
+
+// a device meets its device spec when the version of the device spec that the agent reports as running
+// equals the version rendered to the device by the service
+deviceMeetsDeviceSpec := device.status.config.renderedVersion == device.metadata.annotations.renderedVersion
+
+// a device spec is up-to-date relative to the fleet's device template if both are of the same template version.
+deviceSpecAtFleetSpec := device.metadata.annotations.templateVersion == fleet[device.metadata.owner].spec.templateVersion
+```
+
+The `device.status.updated.???` field contains the current state of the update in progress and can take the following values:
+
+| Update State | Description |
+| ------------ | ----------- |
+| `Preparing` | The agent is validating the desired device spec and downloading dependencies. No changes have been made to the device's configuration yet. |
+| `ReadyToUpdate` | The agent has validated the desired spec, downloaded all dependencies, and is ready to update. No changes have been made to the device's configuration yet. |
+| `ApplyingUpdate` | The agent has started the update transaction and is writing the update to disk. |
+| `Rebooting` | The agent initiated a reboot required to activate the new OS image and configuration. |
+| `ActivatingConfig` | (transient, not reported) The agent is activating the new configuration without requiring a reboot. |
+| `RollingBack` | The agent has detected and error and is rolling back to the pre-update OS image and configuration. |
+| `Updated` | The agent has successfully completed the update and the device is conforming to its device spec. Note that the device's update status may still be reported as `OutOfDate` if the device spec is not yet at the same version as the fleet's device template. |
+| `Cancelled` | The agent has cancelled the update because the desired spec was reverted to the current spec before the update process started. **[TODO: Do we keep downloaded deps?]** |
```

Comment:
Canceled with one L is more common in American English, while the version with two L's is more common in British English.  Personally I like two L's better, but we use American English generally, so we should use one.

Reply:
Note: Canceled isn't implemented yet, so I'll need to remove it from the docs again until then.